cpp/src/Ice/Instance.cpp

```diff
@@ -928,6 +928,12 @@ IceInternal::Instance::initialize(const Ice::CommunicatorPtr& communicator)
 {
     try
     {
+        string programName;
+        if (_initData.properties)
+        {
+            programName = _initData.properties->getIceProperty("Ice.ProgramName");
```

Comment:
It's cleaner to read this property once in this function instead of multiple times. And reading it all the time has the advantage of marking this property as "used".